### PR TITLE
Feature/22 creating services

### PR DESCRIPTION
**Pull request by RobertLi11211**

### CODE REVIEW

File: RideShare/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -2,15 +2,16 @@ import { TestBed, async } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
```
Comment (jurczakn) [inline RideShare/src/app/app.component.spec.ts:2]: App Component should have some tests

File: RideShare/src/app/services/AllDriversServices/all-drivers.service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { User } from '../../models/user.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AllDriversService {
+  /**
+   * Server and port number of the request
+   */
+  readonly port = 'http://localhost:3001';
```
Comment (jurczakn) [inline RideShare/src/app/services/AllDriversServices/all-drivers.service.ts]: move to environment variable

File: RideShare/src/app/services/AllUsersServices/all-users.service.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { User } from 'src/app/models/user.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AllUsersService {
```
Comment (jurczakn) [inline RideShare/src/app/services/AllUsersServices/all-users.service.ts:9]: Comments missing

File: RideShare/src/app/services/AllUsersServices/all-users.service.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  /**
+   * Server and port number of the request
+   */
+  readonly port = 'http://localhost:3001';
```
Comment (jurczakn) [inline RideShare/src/app/services/AllUsersServices/all-users.service.ts]: Move to environment variable

File: RideShare/src/app/services/AllUsersServices/all-users.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      car: null
+    }
+  ];
+
+  it('getAllUsers should get users', () => {
+    service.getAllUsers().subscribe(data => {
+      expect(data).toEqual(UserList);
+    });
+
+    const req = httpMock.expectOne('http://localhost:3001/user');
```
Comment (jurczakn) [inline RideShare/src/app/services/AllUsersServices/all-users.service.spec.ts]: do not hardcode endpoint

File: RideShare/src/app/services/AllDriversServices/all-drivers.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      },
+      car: { seatNumber: 4 }
+    }
+  ];
+  it('getAllDrivers should return drivers', () => {
+    service.getAllDrivers().subscribe(data => {
+      expect(data).toEqual(DriverList);
+    });
+
+    const req = httpMock.expectOne('http://localhost:3001/user?role=DRIVER');
```
Comment (jurczakn) [inline RideShare/src/app/services/AllDriversServices/all-drivers.service.spec.ts]: do not hardcode endpoint

File: RideShare/src/app/services/HousingLocationService/housing-location.service.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * HousingLocationService gets all housing locations
+ *
+ * Imported by:
+ */
+export class HousingLocationService {
+  /**
+   * Server and port number of the request
+   */
+  readonly port = "http://localhost:3003";
```
Comment (jurczakn) [inline RideShare/src/app/services/HousingLocationService/housing-location.service.ts]: move to environment

File: RideShare/src/app/services/HousingLocationService/housing-location.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+      trainingLocation: { trainingLocationName: "USF" }
+    }
+  ];
+
+  it("getHousingLocations() should return data", () => {
+    service.getHousingLocations().subscribe(data => {
+      expect(data).toEqual(housingLocationList);
+    });
+
+    const req = httpMock.expectOne("http://localhost:3003/housing-location");
```
Comment (jurczakn) [inline RideShare/src/app/services/HousingLocationService/housing-location.service.spec.ts]: do not hardcode endpoint

**Review by jurczakn — CHANGES_REQUESTED** [review]
Need to go over

File: RideShare/src/app/app.component.spec.ts (hunk 1 above)
Comment (cmendezlopez25) [reply RideShare/src/app/app.component.spec.ts:2]: App Component is not doing anything by itself, aside from displaying the <app-router>. Should we still have tests for it?

File: RideShare/src/app/services/AllDriversServices/all-drivers.service.ts (hunk 2 above)
Comment (cmendezlopez25) [reply RideShare/src/app/services/AllDriversServices/all-drivers.service.ts]: Changed to environment variable in environments.ts.

File: RideShare/src/app/services/AllUsersServices/all-users.service.ts (hunk 4 above)
Comment (cmendezlopez25) [reply RideShare/src/app/services/AllUsersServices/all-users.service.ts]: Changed to environment variable in environments.ts.

File: RideShare/src/app/services/AllDriversServices/all-drivers.service.spec.ts (hunk 6 above)
Comment (cmendezlopez25) [reply RideShare/src/app/services/AllDriversServices/all-drivers.service.spec.ts]: Changed to environment variable in environments.ts.

File: RideShare/src/app/services/AllUsersServices/all-users.service.spec.ts (hunk 5 above)
Comment (cmendezlopez25) [reply RideShare/src/app/services/AllUsersServices/all-users.service.spec.ts]: Changed to environment variable in environments.ts.

File: RideShare/src/app/services/HousingLocationService/housing-location.service.ts (hunk 7 above)
Comment (cmendezlopez25) [reply RideShare/src/app/services/HousingLocationService/housing-location.service.ts]: Changed to environment variable in environments.ts.

File: RideShare/src/app/services/HousingLocationService/housing-location.service.spec.ts (hunk 8 above)
Comment (cmendezlopez25) [reply RideShare/src/app/services/HousingLocationService/housing-location.service.spec.ts]: Changed to environment variable in environments.ts.

**Review by jurczakn — APPROVED** [review]
good

**Pull request merged**